### PR TITLE
tkt-64152: Fix listing non-standard RELEASE naming schemes

### LINKED ISSUE #596: iocage list -r with non-standard name of the release fails
Make sure to follow and check these boxes before submitting an issue! Thank you.

- [ ] Supply `iocage --version`

0.9.10

- [ ] Supply the commands used, along with any steps to recreate it.

iocage list -r

- [ ] Provide the output from the command you issued.

```python-traceback
# iocage list -r
Traceback (most recent call last):
  File "/usr/local/bin/iocage", line 10, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/iocage/cli/list.py", line 114, in cli
    dataset_type, header, _long, _sort, plugin=plugins, quick=quick)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/iocage.py", line 1200, in list
    exit_on_error=self.exit_on_error).list_datasets()
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_list.py", line 96, in list_datasets
    bases = self.list_bases(ds)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_list.py", line 305, in list_bases
    exit_on_error=self.exit_on_error)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_common.py", line 140, in ioc_sort
    return sort_release(data, split=True)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_common.py", line 298, in sort_release
    rel = float(rel)
ValueError: could not convert string to float: 'r323637M'
```

- [ ] Supply what you expected the result or output to be

```sh-session
# ioc list -r
+------------------+
|    FULL_NAME     |
+==================+
| r323637M-RELEASE |
+------------------+
| r326049M-RELEASE |
+------------------+
| 11.2-RELEASE     |
+------------------+
| r332871M-RELEASE |
+------------------+
```

- [ ] Checked that the problem has not already been fixed on `master` if using
 a stable release.

### PR DESCRIPTION
**Pull request by skarekrow**
Closes #596
FreeNAS Ticket: #64152

### CODE REVIEW

File: iocage_lib/ioc_common.py [hunk 1]
```diff
@@ -393,10 +393,16 @@ def sort_release(releases, split=False, fetch_releases=False):
 
     if split:
         for i, rel in enumerate(releases):
-            rel, r_type = rel.properties["mountpoint"].value.rsplit("/")[
-                -1].split("-", 1)
+            try:
+                rel, r_type = rel.properties["mountpoint"].value.rsplit("/")[
+                    -1].split("-", 1)
+            except ValueError:
```
Comment (william-gr) [inline iocage_lib/ioc_common.py:399]: How can you get ValueError here?

Comment (skarekrow) [reply iocage_lib/ioc_common.py:399]: When the release is named something like `r323637M`.

**Review by william-gr — APPROVED** [review]

**Comment by netchild** [comment]
Hi,

This doesn't fix #596:

# iocage --version
Version 1.0 RELEASE 2018/12

# iocage list -r
Traceback (most recent call last):
  File "/usr/local/bin/iocage", line 10, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/iocage_cli/list.py", line 83, in cli
    dataset_type, header, _long, _sort, plugin=plugins, quick=quick)
  File "/usr/local/lib/python3.6/site-packages/iocage_lib/iocage.py", line 1275, in list
    quick=quick
  File "/usr/local/lib/python3.6/site-packages/iocage_lib/ioc_list.py", line 95, in list_datasets
    bases = self.list_bases(ds)
  File "/usr/local/lib/python3.6/site-packages/iocage_lib/ioc_list.py", line 373, in list_bases
    "list_release", "release", data=datasets)
  File "/usr/local/lib/python3.6/site-packages/iocage_lib/ioc_common.py", line 159, in ioc_sort
    return sort_release(data, split=True)
  File "/usr/local/lib/python3.6/site-packages/iocage_lib/ioc_common.py", line 406, in sort_release
    rel = float(rel)
ValueError: could not convert string to float: 'r323637M'

# ll /iocage/releases
total 3
drwxr-xr-x  3 root  wheel     3B 19 Sep.  2017 r323637M-RELEASE/
drwxr-xr-x  3 root  wheel     3B 22 Nov.  2017 r326049M-RELEASE/
drwxr-xr-x  3 root  wheel     3B 23 Apr.  2018 r332871M-RELEASE/
drwxr-xr-x  3 root  wheel     3B  1 Dez. 22:29 r340754M-RELEASE/
drwxr-xr-x  3 root  wheel     3B  4 Jan. 23:07 r342729M-RELEASE/

Bye,
Alexander.

**Comment by skarekrow** [comment]
It does if they didn't have -RELEASE in the name :D

I'll reopen

**Pull request merged**